trikot-kword/kword-remote-update/src/commonMain/kotlin/com/mirego/trikot/kword/remote/update/internal/RemoteTranslationsFetcher.kt

```diff
@@ -0,0 +1,99 @@
+package com.mirego.trikot.kword.remote.update.internal
+
+import com.mirego.trikot.kword.I18N
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.get
+import io.ktor.http.isSuccess
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.Json
+
+class RemoteTranslationsFetcher(
+    private val baseTranslationsUrl: String?,
+    private val translationsVersion: String?,
+    private val internalCacheWrapper: InternalCacheWrapper
+) {
+    fun fetchRemoteTranslations(
+        i18N: I18N,
+        baseFileName: String,
+        baseMap: MutableMap<String, String>,
+        languageCodes: List<String>
+    ) {
+        baseTranslationsUrl?.let { baseTranslationsUrlVal ->
+            translationsVersion?.let { translationsVersionVal ->
+                if (baseTranslationsUrlVal.isEmpty() || translationsVersionVal.isEmpty()) {
+                    return
+                }
+
+                val sharedHttpClient = HttpClient()
+                CoroutineScope(Dispatchers.Unconfined).launch {
```

Comment:
Unconfined will dispatch the coroutine on the thread the caller used. async will inherit that scope and might probably end up doing IO on the main thread.